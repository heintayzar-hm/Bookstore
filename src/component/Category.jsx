import React from 'react';

// eslint-disable-next-line react/prefer-stateless-function
class Category extends React.Component {
  constructor(props) {
    super();
    this.props = props;
  }

  render() {
    return (
      <>
        <div>
          <button data-testid="check-status" type="button">Check Status</button>
        </div>
      </>
    );
  }
}
export default Category;
