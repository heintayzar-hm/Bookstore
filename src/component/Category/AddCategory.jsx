import React from 'react';
import PropTypes from 'prop-types';

// eslint-disable-next-line react/prefer-stateless-function
class AddCategory extends React.Component {
  constructor(props) {
    super();
    this.props = props;
    this.state = {
      name: '',
    };
  }

  changeHandler = (e) => {
    e.preventDefault();
    this.setState((prevState) => ({
      ...prevState,
      [e.target.name]: e.target.value,
    }));
  }

  submitHandler = (e) => {
    e.preventDefault();
    const { addCategory } = this.props;
    const { name } = this.state;
    if (name.trim()) {
      addCategory(name);
      this.setState({
        name: '',
      });
    }
  }

  render() {
    // eslint-disable-next-line no-unused-vars
    const { name } = this.state;
    return (
      <>
        <article data-testid="form-container" className="py-5 px-10">
          <h2 className="font-important tracking-wider text-[1.25rem] font-bold text-[#888] uppercase pb-3 ">Add New Category</h2>
          <form onSubmit={this.submitHandler} className=" gap-4 grid grid-cols-1 sm:grid-cols-12 ">
            <input className="dark:bg-black dark:text-white sm:col-span-3 p-2 px-3 border focus:outline-none rounded focus:border-blue border-solid tracking-widest text-brown" name="name" onChange={this.changeHandler} data-testid="category-input" type="text" placeholder="Category Name" value={name} required />
            <button className="sm:col-span-2 p-3 bg-blue text-white rounded hover:bg-white hover:text-blue" type="submit">Add Category</button>
          </form>
        </article>
      </>
    );
  }
}

AddCategory.propTypes = {
  addCategory: PropTypes.func.isRequired,
};
export default AddCategory;
