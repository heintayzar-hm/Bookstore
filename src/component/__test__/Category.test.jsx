import { describe, expect, test } from 'vitest';
import { render } from '@testing-library/react';
import Category from '../Category';

describe('Category', () => {
  test('renders the initial state correctly', () => {
    // Arrange

    // Act
    const { container } = render(
      <Category />,
    );
    // Assert
    expect(container).toMatchSnapshot();
  });

  test('Have check status button', () => {
    // Arrange

    // Act
    const { queryByTestId } = render(
      <Category />,
    );
    const button = queryByTestId('check-status');
    // Assert
    expect(button.innerHTML).toBe('Check Status');
  });
});
