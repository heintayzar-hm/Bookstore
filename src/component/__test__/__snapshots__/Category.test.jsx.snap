// Vitest Snapshot v1

exports[`Category > renders the initial state correctly 1`] = `
<div>
  <div>
    <button
      data-testid="check-status"
      type="button"
    >
      Check Status
    </button>
  </div>
</div>
`;
